src/libkomb.py

```diff
@@ -0,0 +1,504 @@
+                    )
+                if st.get("linecap"):
+                    mapbox_style_layer["layout"]["line-cap"] = to_mapbox_expression(
+                        {z: mapbox_linecaps[v] for z, v in st.get("linecap").items()}
+                    )
+                if st.get("linejoin"):
+                    mapbox_style_layer["layout"]["line-join"] = to_mapbox_expression(
+                        st.get("linejoin")
+                    )
+
+                mapbox_style_layer_id += 1
+                mapbox_style_layers.append(mapbox_style_layer)
+            if "fill-color" in st and any(
+                v is not None for v in st.get("fill-color").values()
+            ):
+                mapbox_style_layer = {
+                    "type": "fill",
+                    "filter": mapbox_style_layer_filter,
+                    "layout": {},
+                    "paint": {},
+                    "id": "+".join(build_kepler_hints(conditions, st))
+                    + str(mapbox_style_layer_id),
+                    "source-layer": subject,
+                    "source": "composite",
+                }
+
+                mapbox_style_layer["minzoom"] = sorted(
+                    st.get("fill-color").items(), key=lambda k: k[0]
+                )[0][0]
+                mapbox_style_layer["maxzoom"] = (
+                    sorted(st.get("fill-color").items(), key=lambda k: k[0])[-1][0] + 1
+                )
+
+                if st.get("fill-position", "foreground") == "background":
+                    if "z-index" not in st:
+                        bgpos -= 1
+                        mapbox_style_layer["priority"] = bgpos - 16000
+                    else:
+                        zzz = int(st.get("z-index", 0))
+                        if zzz > 0:
+                            mapbox_style_layer["priority"] = zzz - 16000
+                        else:
+                            mapbox_style_layer["priority"] = zzz - 16700
+                else:
+                    mapbox_style_layer["priority"] = (
+                        int(st.get("z-index", 0)) + 1 + 1000
+                    )
+
+                mapbox_style_layer["paint"]["fill-color"] = to_mapbox_expression(
+                    {z: whatever_to_hex(v) for z, v in st.get("fill-color").items()}
+                )
+
+                if st.get("fill-opacity"):
+                    mapbox_style_layer["paint"]["fill-opacity"] = to_mapbox_expression(
+                        st.get("fill-opacity")
+                    )
+
+                mapbox_style_layer_id += 1
+                mapbox_style_layers.append(mapbox_style_layer)
+            if st.get("text"):
+                mapbox_style_layer = {
+                    "type": "symbol",
+                    "filter": mapbox_style_layer_filter,
+                    "layout": {
+                        "text-font": ["Roboto"],
+                        "symbol-sort-key": ["-", ["to-number", ["get", "population"]]],
```

Comment:
can sort here be by two things? admin_level before population.

Reply:
since features are already sorted in tiles by (admin_level, population) I would rather remove sorting on client side